core/src/blockchain/body_db.rs

```diff
@@ -162,40 +176,53 @@ impl BodyDB {
                 // The order here is important! Don't remove parcel if it was part of enacted blocks as well.
                 retracted.chain(enacted).chain(current_addresses).collect()
             }
-            BlockLocation::Branch => HashMap::new(),
+            BestBlockChanged::None => HashMap::new(),
         }
     }
 
     fn new_transaction_address_entries(
         &self,
-        block: &BlockView,
-        location: &BlockLocation,
+        best_block_changed: &BestBlockChanged,
     ) -> HashMap<H256, Option<TransactionAddress>> {
+        let block_hash = if let Some(best_block_hash) = best_block_changed.new_best_hash() {
+            best_block_hash
+        } else {
+            return HashMap::new()
+        };
+        let block = match best_block_changed.best_block() {
+            Some(block) => block,
+            None => return HashMap::new(),
+        };
+
+        // FIXME: is the block is new best block?
```

Comment:
I can't think of the cases that `block` doesn't become a new best block for now.
Do you have any situation in mind?